graylog2-server/src/main/java/org/graylog/plugins/views/search/export/CommandFactory.java

```diff
@@ -104,12 +103,15 @@ private AbsoluteRange toAbsolute(TimeRange timeRange) {
     }
 
     private ExportMessagesCommand.Builder builderFrom(ResultFormat resultFormat) {
-        ExportMessagesCommand.Builder requestBuilder = ExportMessagesCommand.builder();
-
-        requestBuilder.fieldsInOrder(resultFormat.fieldsInOrder());
+        ExportMessagesCommand.Builder requestBuilder = ExportMessagesCommand.builder()
+                .fieldsInOrder(resultFormat.fieldsInOrder());
 
         if (resultFormat.limit().isPresent()) {
-            requestBuilder.limit(resultFormat.limit().orElseThrow(() -> new IllegalStateException("No value present!")));
+            requestBuilder = requestBuilder.limit(resultFormat.limit().orElseThrow(() -> new IllegalStateException("No value present!")));
+        }
+
+        if (resultFormat.timeZone().isPresent()) {
+            requestBuilder = requestBuilder.timeZone(resultFormat.timeZone().orElseThrow(() -> new IllegalStateException("No time zone present!")));
```

Comment:
```suggestion
        resultFormat.timeZone().ifPresent(requestBuilder::timeZone);
```

The logic of the condition prevents any possible IllegalStateException, so I'd suggest to simplify the expression. And maybe adapt the very same expression with `limit` above?

Reply:
I don't have a strong opinion towards this. I tried to avoid assuming that the builder is mutable, but with the requirements of variables being (effectively) final in closures and therefore allowing no reassignment, it results in slightly clumsy code.

Reply:
Isn't a builder by definition mutable, building immutable objects as a result? I'd prefer to mutate the builder rather than to re-assign the local variable. And we'd avoid the duplicated check if the timezone is present.

Reply:
I am a bit used to immutable builders which we are using in the frontend. As long as we are using AutoValue for builders, we are fine here too.